src/System.Runtime/tests/System/ActivatorTests.cs

```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor)));
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor), nonPublic: true));
+            Assert.IsType<ValueTypeWithDefaultConstructor>(Activator.CreateInstance(typeof(ValueTypeWithDefaultConstructor), nonPublic: false));
+        }
+
+        [Fact]
+        public void CreateInstance_NonPublicClassWithPrivateDefaultConstructor_Success()
+        {
+            // Test caching behaviour by activating multiple times.
+            ClassWithPrivateDefaultConstructor c1 = (ClassWithPrivateDefaultConstructor)Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: true);
+            Assert.Equal(-1, c1.Property);
+
+            ClassWithPrivateDefaultConstructor c2 = (ClassWithPrivateDefaultConstructor)Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: true);
+            Assert.Equal(-1, c2.Property);
+        }
+
+        [Fact]
+        public void CreateInstance_PublicOnlyClassWithPrivateDefaultConstructor_ThrowsMissingMethodException()
+        {
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor)));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: false));
 
+            // Put the private default constructor into the cache and make sure we still throw if public only.
+            Assert.NotNull(Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: true));
+
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor)));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: false));
+        }
+
+        [Fact]
+        public void CreateInstance_NullableType_ReturnsNull()
+        {
+            Assert.Null(Activator.CreateInstance(typeof(int?)));
+        }
+
+        [Fact]
+        public void CreateInstance_NullType_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance((Type)null));
+            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance(null, new object[0]));
+        }
+
+        [Fact]
+        public static void CreateInstance_MultipleMatchingConstructors_ThrowsAmbiguousMatchException()
+        {
             Assert.Throws<AmbiguousMatchException>(() => Activator.CreateInstance(typeof(Choice1), new object[] { null }));
+        }
 
+#if netcoreapp || uapaot
+        [Fact]
+        public void CreateInstance_NotRuntimeType_ThrowsArgumentException()
+        {
+            foreach (Type nonRuntimeType in Helpers.NonRuntimeTypes)
```

Comment:
Can this be a Theory instead?

Reply:
No, due to https://github.com/xunit/xunit/issues/1325